docs/tutorials/p2p.md

```diff
@@ -106,61 +96,52 @@ It pretty simply acknowledges the session request, accepts it, then sends a hand
+				# Get the remote user's ID
+				var message_sender: int = message['remote_steam_id']
 
-			# Print the packet to output
-			print("Packet: %s" % readable_data)
+				# Print the packet to output
+				print("Packet: " + message.payload)
 
-			# Append logic here to deal with packet data
+				# Append logic here to deal with message data. Do NOT use call or call_v here when directly dealing with data without sanatizing your data first. Ideally don't use it at all unless you really know what you're doing.
 	```
 
 === "Godot 4.x"
 
 	```gdscript
-	func read_p2p_packet() -> void:
-		var packet_size: int = Steam.getAvailableP2PPacketSize(0)
+	func read_messages() -> void:
+		var messages: Array = Steam.receiveMessagesOnChannel(0, 1000)
 
-		# There is a packet
-		if packet_size > 0:
-			var this_packet: Dictionary = Steam.readP2PPacket(packet_size, 0)
-
-			if this_packet.is_empty() or this_packet == null:
-				print("WARNING: read an empty packet with non-zero size!")
+		for message in messages
+			if message.is_empty() or message == null:
+				print("WARNING: read an empty message with non-zero size!")
+     		else:
+				message.payload = bytes_to_var(message.payload)
+				# Get the remote user's ID
+				var message_sender: int = this_packet['remote_steam_id']
 
-			# Get the remote user's ID
-			var packet_sender: int = this_packet['remote_steam_id']
+				# Print the packet to output
+				print("Message: " + message.payload)
 
-			# Make the packet data readable
-			var packet_code: PackedByteArray = this_packet['data']
-			var readable_data: Dictionary = bytes_to_var(packet_code)
-
-			# Print the packet to output
-			print("Packet: %s" % readable_data)
-
-			# Append logic here to deal with packet data
+				# Append logic here to deal with message data. Do NOT use call or call_v here when directly dealing with data without sanatizing your data first. Ideally don't use it at all unless you really know what you're doing.
 	```
 
-If the packet size is greater than zero then it will get the sender's Steam ID and the data they sent. The line `bytes2var` (Godot 2x., 3.x) or `bytes_to_var` (Godot 4.x) is very important as it decodes the data back into something you can read and use. After it is decoded you can pass that data to any number of functions for your game.
+If the packet size is greater than zero then it will get the sender's Steam ID and the data they sent. The line `bytes2var` (Godot 2x., 3.x) or `bytes_to_var` (Godot 4.x) is very important as it decodes the data back into something you can read and use. Do NOT use bytes_to_var_with_objects or bytes2var with objects enabled as it can lead to remote code execution. After it is decoded you can pass that data to any number of functions for your game. It is also important to remember to not allow the other users to call functions directly as well with the packets they're sending (or else it may ALSO lead to compromised user security and remote code execution).
```

Comment:
This probably needs re-written with more specifics and examples as to how and why.